src/inner_product_proof.rs

```diff
@@ -51,15 +51,6 @@ impl InnerProductProof {
 
         let mut n = G.len();
 
-        if n == 0 {
-            return InnerProductProof {
-                L_vec: vec![],
-                R_vec: vec![],
-                a: Scalar::zero(),
-                b: Scalar::zero(),
-            };
-        }
-
         // All of the input vectors must have the same length.
```

Comment:
Seems like `if n == 0` check got lost in this merge

Reply:
Oh, I misunderstood your comment. I thought that's the check that you've added.

Why don't we make the API complete and allow zero multipliers, but improve the check in `verification_scalars` to handle zero correctly? Even if no-multiplier case is better handled by schnorr proofs, the user of a higher-level protocol may not even be exposed to the amount of multipliers and can accidentally create constraints without any.

Reply:
Any case where you would make a size-0 IPP you could just prove nothing, so I think that rejecting 0 is fine.

Reply:
In a R1CS, internal 0-IPP would mean verifying linear constraints on a bunch of commitments, so it's implementing basically a schnorr proof, just with a few unnecessary challenges. So while IPP part proves nothing, CS proof is still useful.

Reply:
I mean, 0-multiplier r1cs proof is proving a statement `<W_v, V> + W_c = 0` which is not nothing.